src/utils/service_builder.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2019, Arm Limited, All Rights Reserved
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//          http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use crate::authenticators::simple_authenticator::SimpleAuthenticator;
+use crate::back::{backend_handler::BackEndHandlerBuilder, dispatcher::DispatcherBuilder};
+use crate::front::listener::{ListenerConfig, ListenerType};
+use crate::front::{
+    domain_socket::DomainSocketListenerBuilder, front_end::FrontEndHandler,
+    front_end::FrontEndHandlerBuilder, listener::Listen,
+};
+use crate::key_id_managers::on_disk_manager::{OnDiskKeyIDManagerBuilder, DEFAULT_MAPPINGS_PATH};
+use crate::key_id_managers::{KeyIdManagerConfig, KeyIdManagerType, ManageKeyIDs};
+use crate::providers::{
+    core_provider::CoreProviderBuilder, mbed_provider::MbedProviderBuilder, Provide,
+};
+use parsec_interface::operations_protobuf::ProtobufConverter;
+use parsec_interface::requests::AuthType;
+use parsec_interface::requests::{BodyType, ProviderID};
+use serde::Deserialize;
+use std::path::PathBuf;
+use std::sync::Arc;
+use std::sync::RwLock;
+use std::time::Duration;
+use threadpool::{Builder as ThreadPoolBuilder, ThreadPool};
+
+const VERSION_MINOR: u8 = 0;
+const VERSION_MAJOR: u8 = 1;
+
+#[derive(Deserialize)]
+pub struct CoreSettings {
+    pub thread_pool_size: Option<usize>,
+
+    #[serde(default)]
+    pub idle_listener_sleep_duration: Option<u64>,
+}
+
+#[derive(Deserialize)]
+pub struct ServiceConfig {
+    pub core_settings: CoreSettings,
+    pub listener: ListenerConfig,
+    pub key_manager: KeyIdManagerConfig,
+}
+
+pub struct ServiceBuilder;
+
+impl ServiceBuilder {
+    pub fn build_service(config: &ServiceConfig) -> FrontEndHandler {
```

Comment:
Should it be `build_front_end_handler` instead of `build_service`?

Reply:
I understand that it does build almost everything, but is it the whole service though?

Reply:
Well... it is, all the service components (maybe excluding the listener, if you want to consider that part of the service) are created here, so `build_front_end_handler` is way more misleading than `build_service`

Reply:
(and as the configuration will evolve, I think almost every component will get its own `build` function that will be private to this module)